core/server/master/src/main/java/alluxio/master/meta/DefaultMetaMaster.java

```diff
@@ -306,26 +306,18 @@ public void start(Boolean isPrimary) throws IOException {
             ThreadFactoryUtils.build("DailyMetadataBackup-%d", true)), mUfsManager);
         mDailyBackup.start();
       }
-      if (mState.getClusterID() == INVALID_CLUSTER_ID) {
+      if (mState.getClusterID().equals(INVALID_CLUSTER_ID)) {
         try (JournalContext context = createJournalContext()) {
           String clusterID = java.util.UUID.randomUUID().toString();
           mState.applyAndJournal(context, clusterID);
           LOG.info("Created new cluster ID {}", clusterID);
         }
         if (Boolean.valueOf(ProjectConstants.UPDATE_CHECK_ENABLED)
             && ServerConfiguration.getBoolean(PropertyKey.MASTER_UPDATE_CHECK_ENABLED)) {
-          try {
-            String latestVersion =
-                UpdateCheck.getLatestVersion(mState.getClusterID(), 3000, 3000, 3000);
-            if (!ProjectConstants.VERSION.equals(latestVersion)) {
-              System.out.println("The latest version (" + latestVersion + ") is not the same "
-                  + "as the current version (" + ProjectConstants.VERSION + "). To upgrade "
-                  + "visit https://www.alluxio.io/download/.");
-              mNewerVersionAvailable = true;
-            }
-          } catch (Exception e) {
-            LOG.debug("Unable to check for updates: {}", e.getMessage());
-          }
+          getExecutorService().submit(new HeartbeatThread(HeartbeatContext.MASTER_UPDATE_CHECK,
+              new UpdateChecker(this, mState.getClusterID()),
```

Comment:
`DefaultMetaMaster` already has a method `getClusterID`? we can remove the 2nd parameter to the constructor

Reply:
Good advice~!!!